pkg/ebpf/c/bpf_telemetry.h

```diff
@@ -5,69 +5,62 @@
-
-#define PATCH_TARGET_TELEMETRY -1
-static void *(*bpf_telemetry_update_patch)(unsigned long, ...) = (void *)PATCH_TARGET_TELEMETRY;
-
-#define map_update_with_telemetry(fn, map, args...)                                \
-    ({                                                                             \
-        long errno_ret, errno_slot;                                                \
-        errno_ret = fn(&map, args);                                                \
-        unsigned long err_telemetry_key;                                           \
-        LOAD_CONSTANT(MK_KEY(map), err_telemetry_key);                             \
-        if (errno_ret < 0 && err_telemetry_key > 0) {                              \
-            map_err_telemetry_t *entry =                                           \
-                bpf_map_lookup_elem(&map_err_telemetry_map, &err_telemetry_key);   \
-            if (entry) {                                                           \
-                errno_slot = errno_ret * -1;                                       \
-                if (errno_slot >= T_MAX_ERRNO) {                                   \
-                    errno_slot = T_MAX_ERRNO - 1;                                  \
-                    errno_slot &= (T_MAX_ERRNO - 1);                               \
-                }                                                                  \
-                errno_slot &= (T_MAX_ERRNO - 1);                                   \
-                long *target = &entry->err_count[errno_slot];                      \
-                unsigned long add = 1;                                             \
-                /* Patched instruction for 4.14+: __sync_fetch_and_add(target, 1);
-                 * This patch point is placed here because the above instruction
-                 * fails on the 4.4 verifier. On 4.4 this instruction is replaced
-                 * with a nop: r1 = r1 */                                          \
-                bpf_telemetry_update_patch((unsigned long)target, add);            \
-            }                                                                      \
-        }                                                                          \
-        errno_ret;                                                                 \
+#define MK_MAP_INDX(key) STR(key##_telemetry_key)
+
+#define map_update_with_telemetry(fn, map, args...)                     \
+    ({                                                                  \
+        long errno_ret;                                                 \
+        errno_ret = fn(&map, args);                                     \
+        unsigned long map_index;                                        \
+        LOAD_CONSTANT(MK_MAP_INDX(map), map_index);                     \
+        if (errno_ret < 0) {                                            \
```

Comment:
I see you removed the check on `map_index > 0`. I think that was a guard against collecting all telemetry in the same slot, if the constants are not initialized properly.

Reply:
Fixed